src/app/profile/edit-profile/edit-profile.component.ts

```diff
@@ -22,53 +25,116 @@ export class EditProfileComponent implements OnInit {
     firstName: " ",
     lastName: " ",
     team: " ",
-    skills: []
+    genericSkills: [],
+    domainSkills: [],
+    kbcSkills: []
   }
 
   profileForm: FormGroup;
   availableTeams: teams[];
-  userSkills: Skill[];
-  availableSkills: AvailableSkill[];
+  userGenericSkills: Skill[];
+  userDomainSkills: Skill[];
+  userKbcSkills: Skill[];
+  inputSkillArray: Skill[];
+
+  inputId: number;
+  maxSkillValue: number = 8;
+  typeToEdit: string;
+  availableGenericSkills: AvailableSkill[];
+  availableDomainSkills: AvailableSkill[];
+  availableKbcSkills: AvailableSkill[];
+  domLoaded: boolean = false;
 
   constructor(private dataStorageService: DataStorage,
               private fb: FormBuilder,
-              public dialog: MatDialog,) { }
+              private router: Router,
```

Comment:
Declare the private variable with _..
eg: _router

Reply:
and also readonly